core/src/test/scala/org/broadinstitute/dsde/rawls/webservice/WorkspaceApiServiceSpec.scala

```diff
@@ -1694,23 +1695,78 @@ class WorkspaceApiServiceSpec extends ApiServiceSpec {
+      sealRoute(services.workspaceRoutes) ~>
+      check {
+        assertResult(StatusCodes.OK, response.entity.asString) { status }
+        responseAs[WorkspaceACL].acl should contain (testData.userReader.userEmail.value -> AccessEntry(WorkspaceAccessLevels.Read, false, false))
+      }
   }
 
   it should "not allow a writer with share permissions to give permission above their own access level" in withTestDataApiServicesAndUser("writer-access") { services =>
-    assert(true)
+    import WorkspaceACLJsonSupport._
+    Patch(s"/workspaces/${testData.workspaceToTestGrant.namespace}/${testData.workspaceToTestGrant.name}/acl", httpJson(Seq(WorkspaceACLUpdate(testData.userReader.userEmail.value, WorkspaceAccessLevels.Owner, None)))) ~>
+      sealRoute(services.workspaceRoutes) ~>
+      check {
+        assertResult(StatusCodes.BadRequest) { status }
+      }
+    Get(s"/workspaces/${testData.workspaceToTestGrant.namespace}/${testData.workspaceToTestGrant.name}/acl") ~>
+      sealRoute(services.workspaceRoutes) ~>
+      check {
+        assertResult(StatusCodes.OK, response.entity.asString) { status }
+        responseAs[WorkspaceACL].acl should not contain (testData.userReader.userEmail.value -> AccessEntry(WorkspaceAccessLevels.Owner, false, false))
+      }
   }
 
   it should "not allow a writer with share permissions to alter the permissions of users above their access level" in withTestDataApiServicesAndUser("writer-access") { services =>
-    assert(true)
+    import WorkspaceACLJsonSupport._
+    Patch(s"/workspaces/${testData.workspaceToTestGrant.namespace}/${testData.workspaceToTestGrant.name}/acl", httpJson(Seq(WorkspaceACLUpdate(testData.userOwner.userEmail.value, WorkspaceAccessLevels.Read, None)))) ~>
+      sealRoute(services.workspaceRoutes) ~>
+      check {
+        assertResult(StatusCodes.BadRequest) { status }
+      }
+    Get(s"/workspaces/${testData.workspaceToTestGrant.namespace}/${testData.workspaceToTestGrant.name}/acl") ~>
+      sealRoute(services.workspaceRoutes) ~>
+      check {
+        assertResult(StatusCodes.OK, response.entity.asString) { status }
+        responseAs[WorkspaceACL].acl should contain (testData.userOwner.userEmail.value -> AccessEntry(WorkspaceAccessLevels.ProjectOwner, false, true))
+      }
   }
 
-  it should "allow a user in a group with share permissions to share equal to and below their access level" in withTestDataApiServicesAndUser("writer-access") { services =>
-    assert(true)
+  it should "allow a user in a group with share permissions to share equal to and below their access level" in withTestDataApiServicesAndUser("reader-access") { services =>
+    import WorkspaceACLJsonSupport._
+    Patch(s"/workspaces/${testData.workspaceToTestGrant.namespace}/${testData.workspaceToTestGrant.name}/acl", httpJson(Seq(WorkspaceACLUpdate(testData.userReader2.userEmail.value, WorkspaceAccessLevels.Read, None)))) ~>
```

Comment:
Okay, I understand that now.

The problem I see is that the group membership and permissions are not obvious when looking at this test. You have to trace through the all-in-one test fixture to see it, including understanding that there is a database lookup linking `"reader-access"` to `testData.userReader` because there is no way to navigate that statically. The risk with having these facts so far apart is that it's not obvious how changes in `TestDriverComponent` might affect which tests. For example, I committed tests related to bucket usage that use `"reader-access"` on the same day that you added that user to a group. So, while I was doing development, the user wasn't in a group but, by the time both of our branches made it to develop, the user was. Thankfully, it doesn't make a difference for my tests, but I think it illustrates the danger that I talked about at the cross-team tech meeting a few weeks ago.

Can the username be changed to make the user specifics obvious, such as `"reader-access-group-can-share"`? Or maybe even introduce a new user with that name, leaving `"reader-access"` as it was.